### PR TITLE
[TextInput] Add a default max length based on input type 

### PR DESCRIPTION
**Pull request by tanner-ricks**
Added overridable default max length to inputs across the application. This consisted of creating a wrapper component for accomplishing this. Additionally, added/updated e2e tests to verify the fuinctionality.

## Changes

- src
  - components
    - TextInput.tsx: Added this to work as a component wrapper for the design-system TextInput component. This way the character limit defaults can be enforced anywhere this component is used.
    - InputEntry.tsx: Updated this to use the new TextInput wrapper and pass relevant required fails.
  - pages
    - PointOfContact
      - index.tsx: Added relevant character limits that were different than the default
    - ProfileForm
      - CreateProfileForm
        - AddFinancialInstitution.tsx:  Added relevant character limits that were different than the default
      - Step1Form
        - Step1FormInfoFieldGroup.tsx:  Removed unnecessary character limits since there is now a default
    - types
      - formTypes.ts: Updated default character limits to new variable name
    - utils
      - constants.ts: Updated and created input character limit constants for use
- e2e
  - pages
    - filing-app
      - complete-user-profile
        - checkCupFormErrors.spec.tsx: Added new test to verify character limits
      - point-of-contact
        - checkPocFormErrors.spec.tsx: Added new test to verify character limits
    - shared-lending-platform
      - NonAssociatedUserProfile.spec.tsx: Updated test to verify all character limits
      - UpdateInstitutionProfile.spec.tsx: Added test to verify character limits
  - utils
    - inputValidators.tsx: Added helper functions to unify input content validation

## How to test this PR

1. Pull **986-default-max-length** in **sbl-frontend**
2. Verify that **TextInput.tsx** in **sbl-frontend** has no errors
3. Start the stack
4. Run the e2e tests
5. Verify that the e2e tests pass

### CODE REVIEW

**Review by meissadia — CHANGES_REQUESTED** [review]
Can you please double check your tests? I'm getting consistent failures in the unicode and length tests:
Screenshot 2024-10-10 at 12 16 34 PM

**Comment by tanner-ricks** [comment]
> Can you please double check your tests? I'm getting consistent failures in the unicode and length tests: Screenshot 2024-10-10 at 12 16 34 PM

Can't reproduce and can't see the nature of the failures with the information given.

File: e2e/pages/shared-lending-platform/UpdateInstitutionProfile.spec.ts [hunk 1]
```diff
@@ -191,3 +194,111 @@ test('Update Institution Profile Page', async ({
...
+    });
+    await assertTextInput(page, '#top_holder_lei', {
+      isLocator: true,
+      fill: controlUnicode,
+      expected: expectedValues.topLeiField,
+      unexpected: unexpectedValues.topLeiField,
+    });
+    await assertTextInput(page, '#top_holder_rssd_id', {
+      isLocator: true,
+      // fill: controlUnicode,
```
Comment (billhimmelsbach) [inline e2e/pages/shared-lending-platform/UpdateInstitutionProfile.spec.ts:298]: NIT: Old comment to remove?

File: src/utils/constants.ts [hunk 2]
```diff
@@ -11,8 +11,15 @@ export const Ten = 10;
 export const ITEMS_PER_PAGE = 20;
 export const Thirty = 30;
 export const Hundred = 100;
-export const inputCharLimit = 254;
+
+export const DefaultInputCharLimit = 255;
```
Comment (billhimmelsbach) [inline src/utils/constants.ts:15]: I appreciate you adding to this list. Might be something to import from [regtech-regex](https://github.com/cfpb/regtech-regex) someday in the future potentially, maybe after #811 is worked on.

**Review by billhimmelsbach — APPROVED** [review]
Looks good! I think with [this PR](https://github.com/cfpb/sbl-frontend/pull/1023) all tests are passing (see screenshot).

I appreciate the expected/unexpectedValues you use here in the tests and making it easy to make those assertions in the future with inputValidator, and making a text input that has the correct max chars based on type. Thanks!

Screenshot 2024-10-22 at 10 17 42 AM

**Pull request merged**